src/skmultiflow/meta/learn_nse.py

```diff
@@ -0,0 +1,286 @@
+                it has too low accuracy on the dataset.
+        Returns
+        -------
+        LearnNSE
+            self
+        """
+        N, _ = X.shape
+        if self.classes is None:
+            if classes is None:
+                raise RuntimeError("Should pass the classes in the first partial_fit call")
+            else:
+                self.classes = classes
+
+        for i in range(N):
+            self.X_batch.append(X[i])
+            self.y_batch.append(y[i])
+            mt = len(self.y_batch)
+
+            if mt == self.period:
+                self.X_batch = np.array(self.X_batch)
+                self.y_batch = np.array(self.y_batch)
+
+                classifier = cp.deepcopy(self.base_estimator)
+
+                if len(self.ensemble) > 0:
+                    # Compute the error of the existing ensemble on the new data
+                    votes = self.predict(self.X_batch)
+
+                    et = np.sum(votes != self.y_batch) / mt
+
+                    # Update and normalize instance weights
+                    self.instance_weights = np.ones(mt) / mt
+                    self.instance_weights[votes == self.y_batch] = et / mt
+
+                    # normalize instance weights (distribution)
+                    self.instance_weights = self.instance_weights / np.sum(self.instance_weights)
+
+                    # Train base classifier with Dt
+                    self.__fit(classifier, self.X_batch, self.y_batch, sample_weight=self.instance_weights,
+                               classes=self.classes)
+                    # classifier.fit(self.X_batch, self.y_batch, sample_weight=self.instanceWeights)
+
+                else:
+                    # First run! train the classifier on the instances with the same weight
+                    self.instance_weights = np.ones(mt) / mt
+
+                    # classifier.fit(self.X_batch, self.y_batch, sample_weight=self.instanceWeights)
+                    self.__fit(classifier, self.X_batch, self.y_batch, sample_weight=self.instance_weights,
+                               classes=self.classes)
+
+                self.ensemble.append(classifier)
+                self.bkts.append([])
+                self.wkts.append([])
+                self.ensemble_weights = []
+
+                t = len(self.ensemble)
+                max_error = None
+                error_index = None
+
+                # Evaluate all existing classifiers on the new dataset
+                for k in range(1, t + 1):
+                    pred = self.ensemble[k - 1].predict(self.X_batch)
+                    ekt = np.sum(self.instance_weights[pred != self.y_batch])
+
+                    if k == t and ekt > 0.5:
+                        # Generate a new classifier
+                        classifier = cp.deepcopy(self.base_estimator)
+                        self.__fit(classifier, self.X_batch, self.y_batch, sample_weight=self.instance_weights,
```

Comment:
The new classifier is not fitted at this point (in the MOA implementation). It seems strange so we need to confirm in the paper.

Reply:
If I don't fit it, it throws the error the classifier is not fitted when I run to the next iteration.

Reply:
![screenshot from 2019-02-25 13-00-47](https://user-images.githubusercontent.com/9668858/53336046-6d1cdc00-38fd-11e9-8613-910c3de09e92.png)
Here is from the paper. The new classifier needs to be trained

Reply:
This makes sense to me. Now we have to consider that our implementation will provide different results once this part of the code is reached (which in my experiments does).

We can use tests on a smaller scale to validate our implementation.